src/app/src/main/java/com/example/bookmark/MyBookDetailsActivity.java

```diff
@@ -0,0 +1,160 @@
+package com.example.bookmark;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+/**
+ * This activity shows the details of a book. Depending on the
+ * status of the book the user can then take some action. A user
+ * can also navigate to the edit book activity from here.
+ *
+ * TODO: Add more to these? Classes/Listeners?
+ * @author Mitch Adam.
+ */
+public class MyBookDetailsActivity extends AppCompatActivity {
+    // TODO: Figure out the back navigation
+
+    String isbn;
+    String title;
+    String author;
+    String description;
+    String status;
+    //Image image;
+
+    private TextView titleEditText;
+    private TextView authorEditText;
+    private TextView isbnEditText;
+    private TextView descriptionEditText;
+    private ImageView imageView;
+    private TextView statusText;
+
+    private Button actionButton;
+
+    private View.OnClickListener manageRequestsListener = new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            manageRequests();
+        }
+    };
+
+    private View.OnClickListener giveBookListener = new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            giveBook();
+        }
+    };
+
+    private View.OnClickListener receiveBookListener = new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            receiveBook();
+        }
+    };
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_my_book_details);
+        getSupportActionBar().setTitle("Book Details");
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        Intent myIntent = getIntent(); // gets the previously created intent
+        isbn = myIntent.getStringExtra("ISBN");
+
+        titleEditText = findViewById(R.id.book_details_title_text);
+        authorEditText = findViewById(R.id.book_details_author_text);
+        isbnEditText = findViewById(R.id.book_details_isbn_text);
+        descriptionEditText = findViewById(R.id.book_details_description_text);
+        imageView = findViewById(R.id.book_details_book_image);
+        statusText = findViewById(R.id.book_details_book_status_text);
+
+        actionButton = findViewById(R.id.book_details_action_btn);
+
+        getBookDetailsFromISBN();
+        fillBookDetails();
+        configureActionButton();
+    }
+
+    private void getBookDetailsFromISBN() {
+        // TODO: Get details from firebase
```

Comment:
I'll try to get the basics of the FirebaseProvider implemented in the next few days so that you can integrate with it.